### ISSUE TITLE
Improve searching for posts and discussions

### ISSUE THREAD

**Issue by KarimJedda** [Premiurly/polkassembly#330]
Searching for things in the UI is currently using Google Search, which doesn't always return the correct results since it is dependent on when Google indexes the website and new entries. 

An example. Searching for "Proposal: ink! design audit by OpenZeppelin" yields
<img width="1130" alt="Screenshot 2023-01-11 at 08 54 25" src="https://user-images.githubusercontent.com/5029053/211773309-1a35d04a-cd15-4959-bfbb-9ef88cc32d49.png">
which can be confusing. Ideally the search could show exact title matches, search in discussions only etc... 

Since Polkassembly is using Hasura in the backend, one could leverage that search first. 
<img width="1444" alt="Screenshot 2023-01-11 at 10 32 54" src="https://user-images.githubusercontent.com/5029053/211773675-8e360389-cb3c-4caf-867c-8851a3ec531f.png">

Some proposed improvements:
- Default to full text search on title, content for posts and discussions 
- Make it visible what type of object (post, discussion, ...) is returned in the search result list

My javascript skills are not the yellow of the egg so I'm not sure I can properly contribute, however I hope this brings the issue to your attention in a constructive manner.

**Comment by Tbaut** [comment]
I second this. Searching is pretty hard right now. The fact that search engines don't actually index React pages well makes it so that you can't actually find some older proposals.

Try it out with a proposal I did a couple weeks ago. It has a unique searchable name "Multix". There was a discussion, a treasury proposal and a motion with this title. 

The search yield 1 result that brings to an un-useful discussion overview rather than the discussion itself. Searching any search engine with "multix polkassembly" brings to the dashboard as well. I think some research to make sure search engine can index things properly (by having fixed URLs) would help tremendously.

**Comment by alphainfinitus** [comment]
We are pleased to announce that the issue regarding the improvement of searching for posts and discussions on Polkassembly has been successfully addressed and closed with https://github.com/polkassembly/polkassembly/pull/333.

The following improvements that have been implemented:

- Enhanced search algorithm: We have upgraded the search functionality to ensure more accurate and relevant results. The new algorithm takes into account exact title matches, allowing users to find posts and discussions more efficiently.

- Customized search options: We have implemented additional search filters that enable users to narrow down their searches. These options include searching within discussions only, filtering by specific categories, and more.

- Real-time indexing: To eliminate the delays caused by Google indexing, we have implemented a real-time indexing system. This ensures that new entries are promptly available for search, improving the overall user experience.

We appreciate your patience and valuable feedback during the resolution of this issue. Your suggestions and insights have been instrumental in driving these improvements.

Thank you for your continued support and contribution to the Polkassembly ecosystem.
